pkg/apis/pipeline/v1alpha1/verificationpolicy_types.go

```diff
@@ -90,6 +90,13 @@ type KeyRef struct {
 	// Data contains the inline public key.
 	// +optional
 	Data string `json:"data,omitempty"`
+	// KMS contains the KMS url of the public key
+	// Supported formats differ based on the KMS system used.
+	// One example of a KMS url could be:
+	// gcpkms://projects/[PROJECT]/locations/[LOCATION]>/keyRings/[KEYRING]/cryptoKeys/[KEY]/cryptoKeyVersions/[KEY_VERSION]
+	// Note that the KMS is not supported yet.
+	// +optional
+	KMS string `json:"kms,omitempty"`
```

Comment:
Consider using a pointer for an optional type : https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
I cannot find a pointer string being used in our api, I think we should keep it consistent?

Reply:
The API conventions are a bit confusing here, it says "Using a pointer allows distinguishing unset from the zero value for that type" but also "we ask that pointers always be used with optional fields that do not have a built-in nil value" (assuming that "nil value" means "zero value" in this context).

I agree with Yongxuan here, I think having an empty string implies that the URL is unset, not that the URL is "", so there's not a strong need to distinguish.